Libs/iSENSE_API/Source/QueueUploaderView.m

```diff
@@ -462,8 +493,53 @@ - (void) loginAndUploadWithUsername:(NSString *)usernameInput withPassword:(NSSt
     });
 }
 
+// Logs into iSENSE with a contributor key and attempts to upload data
+- (void)loginAndUploadWithContributorKey:(NSString *)contribKey {
+
+    UIAlertView *message = [self getDispatchDialogWithMessage:@"Validating key..."];
+    [message show];
+
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+
+        QDataSet *firstDataSet = (QDataSet *) [dataSaver getDataSet];
+        int pid = [firstDataSet projID].intValue;
+
+        bool isKeyValid = [api validateKey:contribKey forProject:pid];
+
+        dispatch_async(dispatch_get_main_queue(), ^{
+
+            if (isKeyValid) {
```

Comment:
I believe you know what I'm going to say about this if block. Should be fairly easy, since there is already a return on the 'Invalid' case